### DISCUSSION TITLE
does getKnownCompilers work nicely with remote SSH?

### DISCUSSION THREAD

**Discussion by starball5** [microsoft/vscode-cpptools#12165] [Q&A]
I ran into a user on SO who said compiler detection is working for them on WSL, but not when using remote SSH. Is there any particular support for the remote SSH scenario in `getActiveCompiler`?

**Comment by sean-mcmanus** [comment]
@starball5 It should work the same. Do they have a compiler installed on the remote SSH machine?

**Reply by starball5** [reply]
I'm not sure. I'll ping that user to join this discussion. (the question is [here](https://stackoverflow.com/q/78262644/11107541))

**Comment by Colengms** [comment]
From the perspective of the extension, there is no difference between running locally vs remotely on the same hardware.  Both involve installing the same build flavor of the extension. The same code is executed regardless of whether local or remote. 
Differences in behavior would be due to external factors, such as the privileges of the account the code is being run under.  Issues with the C/C++ Extension that repro only via remote and not locally on the same hardware, under the same account, are likely related to VS Code itself and its support for remoting.
